### PR TITLE
Do not create a Solid account folder if OIDC user creation fails

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Fixes issue #387.
Also adds some doc strings.

### CODE REVIEW

File: lib/identity-provider.js [hunk 1]
```diff
@@ -524,15 +531,77 @@ IdentityProvider.prototype.getGraph = function (uri, callback) {
...
-IdentityProvider.prototype.post = function (req, res, next) {
-  if (!req.body) {
+/**
+ * Sends a request to the OIDC Provider's Users API endpoint, to create a new
+ * user record with the provider.
+ * @method createOIDCUser
+ * @param oidcRpClient {OidcRpClient} Local/trusted OIDC client
+ * @param webId {String} WebID URL of the new user to be created
+ * @param options {Object} User options hashmap
+ * @param options.password {String} User's signin password. NOTE: Must be 8+
```
Comment (nicola) [inline lib/identity-provider.js]: this should be optional too, we can of course pass a setting flag that is `--auth-password` or similar, we shouldn't enforce password as auth schema

File: lib/identity-provider.js [hunk 2]
```diff
@@ -524,15 +531,77 @@ IdentityProvider.prototype.getGraph = function (uri, callback) {
...
+ *   characters, mix of alpha and numeric
+ * @param [options.email] {String} User's email address (for recovery etc)
+ * @param [options.name] {String} User's name
+ * @return {Promise}
+ */
+IdentityProvider.prototype.createOIDCUser =
```
Comment (nicola) [inline lib/identity-provider.js]: createOIDCUser shouldn't be in the Identity provider package (maybe put that in the oidc one?)

Comment (dmitrizagidulin) [inline lib/identity-provider.js]: > createOIDCUser shouldn't be in the Identity provider package

Yeah, I agree. Should we factor out the tls-related user creation files somewhere too?

File: lib/identity-provider.js (hunk 1 above)
Comment (dmitrizagidulin) [inline lib/identity-provider.js]: > this should be optional too

Not sure I undersrtand. It's not optional, currently. (in our oidc integration)

**Comment by dmitrizagidulin** [comment]
@nicola - removed the oidc-related user code for now. Ready for review.

**Comment by dmitrizagidulin** [comment]
(The fix to #387 has been merged into the `dz_oidc` branch, however)

**Pull request merged**